code/modules/admin/verbs/map_export.dm

```diff
@@ -0,0 +1,284 @@
+	for(var/item in future_string)
+		if(!first_entry)
+			. += ", "
+		if(future_string[item])
+			. += hashtag_newlines_and_tabs("[item] = [future_string[item]]", list("{"="", "}"="", "\""="", ";"="", ","=""))
+		else
+			. += hashtag_newlines_and_tabs("[item]", list("{"="", "}"="", "\""="", ";"="", ","=""))
+		first_entry = FALSE
+	. += ")"
+
+/**
+ *Procedure for converting a coordinate-selected part of the map into text for the .dmi format
+ */
+/proc/write_map(
+			minx,
+			miny,
+			minz,
+			maxx,
+			maxy,
+			maxz,
+			save_flag = SAVE_ALL,
+			shuttle_area_flag = SAVE_SHUTTLEAREA_DONTCARE,
+			list/obj_blacklist = list()
+)
+
+	var/width = maxx - minx
+	var/height = maxy - miny
+	var/depth = maxz - minz
+
+	//Step 0: Calculate the amount of letters we need (26 ^ n > turf count)
+	var/turfsNeeded = width * height
+	var/layers = FLOOR(log(GLOB.save_file_chars.len, turfsNeeded) + 0.999,1)
+
+	//Step 1: Run through the area and generate file data
+	var/list/header_chars = list() //The characters of the header
+	var/list/header_dat = list() //The data of the header, lines up with chars
+	var/header = "" //The actual header in text
+	var/contents = "" //The contents in text (bit at the end)
+	var/index = 1
+	for(var/z in 0 to depth)
+		for(var/x in 0 to width)
+			contents += "\n([x + 1],1,[z + 1]) = {\"\n"
+			for(var/y in height to 0 step -1)
+				CHECK_TICK
+				//====Get turfs Data====
+				var/turf/place = locate((minx + x), (miny + y), (minz + z))
+				var/area/location
+				var/list/objects
+				var/area/place_area = get_area(place)
+				//If there is nothing there, save as a noop (For odd shapes)
+				if(!place)
+					place = /turf/template_noop
+					location = /area/template_noop
+					objects = list()
+				//Ignore things in space, must be a space turf
+				else if(istype(place, /turf/open/space) && !(save_flag & SAVE_SPACE))
+					place = /turf/template_noop
+					location = /area/template_noop
+				//Stuff to add
+				else
+					location = place_area.type
+					objects = place
+					place = place.type
+				//====Saving shuttles only / non shuttles only====
+				var/is_shuttle_area = istype(location, /area/shuttle)
+				if((is_shuttle_area && shuttle_area_flag == SAVE_SHUTTLEAREA_IGNORE) || (!is_shuttle_area && shuttle_area_flag == SAVE_SHUTTLEAREA_ONLY))
+					place = /turf/template_noop
+					location = /area/template_noop
+					objects = list()
+				//====For toggling not saving areas and turfs====
+				if(!(save_flag & SAVE_AREAS))
+					location = /area/template_noop
+				if(!(save_flag & SAVE_TURFS))
+					place = /turf/template_noop
+				//====Generate Header Character====
+				var/header_char = calculate_tgm_header_index(index, layers)	//The characters of the header
+				var/current_header = "(\n"										//The actual stuff inside the header
+				//Add objects to the header file
+				var/empty = TRUE
+				//====SAVING OBJECTS====
+				if(save_flag & SAVE_OBJECTS)
+					for(var/obj/thing in objects)
```

Comment:
the way you reach this code means that objects is an area.
Which means this is an `in world` for loop.

Reply:
I don't quite understand what changes you are expecting here, could you tell me more accessibly?

Reply:
Why though? I may be missing something, but objects is actually a single turf (probably should change name?) and you look through objects inside said turf. Why do you need to iterate every object in game? Or am I missing something?